src/vanilla.ts

```diff
@@ -41,10 +41,17 @@ export default function create<TState extends State>(
   const setState: SetState<TState> = (partial, replace) => {
     const nextState = typeof partial === 'function' ? partial(state) : partial
     if (nextState !== state) {
+      const previousState = state
       state = replace
         ? (nextState as TState)
         : Object.assign({}, state, nextState)
-      listeners.forEach((listener) => listener(state))
+      listeners.forEach((listener) => {
+        try {
+          listener(state, previousState)
+        } catch (e) {
+          console.error(e)
+        }
```

Comment:
```suggestion
        listener(state, previousState)
```

We don't need to catch errors in vanilla.js. cc @drcmda